### ISSUE TITLE
Escape in Google Chrome moves the cursor left by one

### ISSUE THREAD

**Issue by ericauld** [godbout/kindaVim.docs#152]
I noticed this a couple days ago. In going from insert mode to normal mode, the cursor moves to the left by one char.

Chrome Version 108.0.5359.124 (Official Build) (arm64)
Macbook Pro 14", 2021
Apple M1 pro chip
MacOS Ventura 13.1

**Comment by godbout** [comment]
yeah, this one unfortunately is going to be almost impossible to get correct. in a perfect world application would support the macOS Accessibility correctly, so that we could make calculations and act on the text with total precision. this is what happens with kV when it has access to Accessibility data, and then uses the Accessibility Strategy. (well, almost with total precision, because there's a lot of info needed, and one incorrect may be enough to fuck things up a little bit.) when kV can't have access to the Accessibility data, it reverts to basic key remapping. so basically `b` is translated by `option leftArrow`. in most cases it "works". in the case you're referring it doesn't. the issue is that because we have no access to the Accessibility data, we can't know if we're at the end of the text, or at the beginning, etc. the only thing we can do is detect if we're dealing with a text field—and then translate `b` by `option leftArrow`—or another UI Element—and then translate by just `leftArrow`.

but now, for Chrome especially, it used to support the macOS Accessibility. but a bug was introduced in May (see https://github.com/godbout/kindaVim.docs/issues/72). i reported it, they fixed it in August, and it's been deployed like two weeks ago (see https://bugs.chromium.org/p/chromium/issues/detail?id=1325643). i have tested it, and it's still not really good. there's an issue with Accessibility data on the last line of a text in Chrome. so i'm still pondering whether i should update or not The Wizard regarding Chrome.

maybe, have a look and see what you think. remove Chrome from the `Key Mapping` Family and let kV handles the automatic switch between using the Accessibility Strategy and the Keyboard Strategy. see if it seems better for you, and let me know. i'll test more myself and see if it's worth updating The Wizard.

**Comment by ericauld** [comment]
OK, will try and let you know. Thanks.

**Comment by ericauld** [comment]
So far I haven't noticed any difference. 

Btw, is it expected behavior that W and B don't work in Chrome (before or after I made the switch)

**Comment by godbout** [comment]
> So far I haven't noticed any difference.

it's probably because in some parts kV will still use the `Keyboard Strategy` because it cannot read the text. so basically it automatically switches to the `Key Mapping` Family. but there are places where you would see the changes. for example in GitHub comments, where kV would use the `Accessibility Strategy`. but see video below. i've tested more, and while the devs fixed one issue, there's now many others. so kV's calculations are wrong, because the data provided by Chrome are wrong. i've been dealing with them for more than a year. they keep breaking stuff and it takes six months to get the fixes implemented so i'm gonna stop. i'll update The Wizard's description about Chrome and will tell users to use the `Key Mapping` Family once and for all.

> Btw, is it expected behavior that W and B don't work in Chrome? (Both before or after I made the switch)

you're gonna see it working in the video. to be able to calculate the limits of s **WORD**, kV needs to have access to all the data. in the case of Chrome 1) forcing the Key Mapping Family prevents from reading the data 2) not forcing the Key Mapping Family makes kV rely on wrong data, which gives a completely inconsistent experience. so yeah, for Chrome, no `W` nor `B`, sorry. as long as they don't get their AX support proper, we can't do much. (my guess is they will probably never.)

https://user-images.githubusercontent.com/121373/209457778-fce71003-990a-4b83-b385-de66c91f5645.mp4

kV is complex hard work, but unfortunately there's still a big part that i don't control, which is how apps support the macOS Accessibility that is needed to do the magic. when it comes to Chrome and Electron apps, we're unfortunately entering the world of thoughts and prayers.

**Comment by godbout** [comment]
in the video you can see at the beginning, on google.com, sometimes the block cursor appears, sometimes it doesn't. that is because kV is in Auto Mode, but the data returned by Chrome is shit. so sometimes kV switches to using the Accessibility Strategy, sometimes it switches to the Keyboard Strategy, **within the same input**. that should not happen within the same input. the problem is that depending where you are within the input, Chrome returns wrong data (in this case they're mixing line number with text length. pretty bad).

then when i deal with the GitHub comment, you can see that `j` goes further down the current line rather than going one line down. same issue. wrong data returned, mixing line numbers with word length.

because Chrome and Electron decided to do their own frameworks so that they can push their work onto any OS, they have to deal with their own AX support manually. their AX team is small, crawling under work, and obviously it's not a Google priority, so you end up with stuff breaking over and over again.
